docs/user/customization.md

```diff
@@ -64,4 +64,106 @@ An example `worker.ign` is shown below. It has been modified to increase the HTT
 }
 ```
 
+## Install time customization for machine configuration
+
+In most cases, user applications should be run on the cluster via Kubernetes workload objects (e.g. daemonset, deployment, etc). For example, if a user wants to run a logging agent on all hosts, in general we would want to steer them towards deploying it as a daemonset object. However, there may be some cases where these workloads need to be executed prior to the node joining the Kubernetes cluster. A specific example that was given from a customer call was that due to a compliance mandate "they must run auditing the second the operating system comes up". In this case, the admin would need the ability to define a custom section of Ignition config (containing the systemd unit that runs the auditing container).
+
+The configuration of machines in OpenShift is controlled using [MachineConfigs][machine-config] and what configuration is applied to a machine in the OpenShift cluster is based on the [MachineConfigPools][machine-config-pool]. To allow customization of machine configuration which is not possible as Day 2 operation, the installer allows users to bring their own custom `machineconfig` objects.
+
+1. `openshift-install create manifests --dir $INSTALL_DIR`
```

Comment:
nit: `--dir` is an `openshift-install`-level option, so I prefer `openshift-install --dir $INSTALL_DIR create manifests`.  That also matches our existing examples:

```console
$ git describe
unreleased-master-450-gdf201360b
$ git grep -- --dir README.md docs
README.md:The best thing to do is always pass the `--dir` argument to `install` and `destroy`.
docs/user/overview.md:openshift-install --dir=initial create install-config
docs/user/overview.md:openshift-install --dir=cluster-0 create cluster
docs/user/overview.md:At any point before running `destroy cluster`, `install-config.yaml` can be regenerated by running `openshift-install --dir=cluster-0 create install-config`.
docs/user/overview.md:openshift-install --dir=cluster-1 create manifests  # warning: this target is unstable
docs/user/overview.md:openshift-install --dir=cluster-1 create cluster
```